content/hacking-atom/sections/creating-a-grammar.md

```diff
@@ -3,203 +3,292 @@ title: Creating a Grammar
+Once you have created a parser, you need to publish it to [the NPM registry](https://npmjs.com) to use it in Atom. To do this, make sure you have a `name` and `version` in your parser's `package.json`:
 
-{{#tip}}
+```js
+{
+  "name": "tree-sitter-mylanguage",
+  "version": "0.0.1",
+  // ...
+}
+```
 
-**Tip:** Grammar packages should start with _language-_.
+then run the command `npm publish`.
 
-{{/tip}}
+#### The Package
 
-The default package template creates a lot of folders that aren't needed for grammar packages. Go ahead and delete the `keymaps`, `lib`, `menus`, and `styles` folders. Furthermore, in `package.json`, remove the `activationCommands` section. Now create a new folder called `grammars`, and inside that a file called `flight-manual.cson`. This is the main file that we will be working with - start by populating it with a [boilerplate template](https://gist.github.com/DamnedScholar/622926bcd222eb1ddc483d12103fd315). Now let's go over what each key means.
+Once you have a Tree-sitter parser that is available on npm, you can use it in your Atom package. Packages with grammars are usually named starting with _language_. You'll need a folder with a `package.json`, a `grammars` subdirectory, and a single `json` or `cson` file in the `grammars` directory, which can be named anything.
 
-`scopeName` is the root _scope_ of your package. This should generally describe what language your grammar package is highlighting; for example, `language-javascript`'s `scopeName` is `source.js` and `language-html`'s is `text.html.basic`. Name it `source.flight-manual` for now.
+```
+language-mylanguage
+├── LICENSE
+├── README.md
+├── grammars
+│   └── mylanguage.cson
+└── package.json
+```
 
-`name` is the user-friendly name that is displayed in places like the status bar or the grammar selector. Again, this name should describe what the grammar package is highlighting. Rename it to `Flight Manual`.
+#### The Grammar File
 
-`fileTypes` is an array of filetypes that `language-flight-manual` should highlight. We're interested in highlighting the Flight Manual's Markdown files, so add the `md` extension to the list and remove the others.
+The `mylanguage.cson` file specifies how Atom should use the parser you created.
 
-`patterns` contains the array of regex patterns that will determine how the file is tokenized.
+#### Basic Fields
 
-#### Adding Patterns
+It starts with some required fields:
 
-To start, let's add a basic pattern to tokenize the words `Flight Manual` whenever they show up. Your regex should look like `\bFlight Manual\b`. Here's what your `patterns` block should look like:
 ```coffee
-'patterns': [
-  {
-    'match': '\\bFlight Manual\\b'
-    'name': 'entity.other.flight-manual'
-  }
-]
+id: 'mylanguage'
+name: 'MyLanguage'
```

Comment:
Since this is human-readable, should it include spaces? If it cannot or should not include spaces, we should probably specify that it should not.